source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -759,17 +759,18 @@ ClientContextImpl::ClientContextImpl(Stats::Scope& scope,
     for (auto& ctx : tls_contexts_) {
       int rc = SSL_CTX_set_alpn_protos(ctx.ssl_ctx_.get(), &parsed_alpn_protocols_[0],
                                        parsed_alpn_protocols_.size());
-      RELEASE_ASSERT(rc == 0, "");
+      RELEASE_ASSERT(rc == 0, Utility::getLastCryptoError().value_or(""));
     }
   }
 
   if (!config.signingAlgorithmsForTest().empty()) {
-    const uint16_t sigalgs = parseSigningAlgorithmsForTest(config.signingAlgorithmsForTest());
-    RELEASE_ASSERT(sigalgs != 0, "");
+    const auto sigalgs = parseSigningAlgorithmsForTest(config.signingAlgorithmsForTest());
+    RELEASE_ASSERT(sigalgs.has_value(), fmt::format("unsupported signing algorithm {}",
+                                                    config.signingAlgorithmsForTest()));
 
     for (auto& ctx : tls_contexts_) {
-      int rc = SSL_CTX_set_verify_algorithm_prefs(ctx.ssl_ctx_.get(), &sigalgs, 1);
-      RELEASE_ASSERT(rc == 1, "");
+      int rc = SSL_CTX_set_verify_algorithm_prefs(ctx.ssl_ctx_.get(), &sigalgs.value(), 1);
```

Comment:
Why is there a code change here? I think we shouldn't be doing this in this PR?

Reply:
Reverted.